### ISSUE TITLE
Mash dist is binary!!!

### ISSUE THREAD

**Issue by aperrin** [marbl/Mash#123]
Hi!

I have a big problem with last mash version (2.2).
When I want to calculate distance matrix between 2 genomes, it returns a binary matrix (p-value is either 0 or 1). 
I tried with a fake genome (10k bp), and it works (I have float values). But with real bacterial genomes, it returns a binary matrix.

I attach the files I used for these tests. Here are the results I get with `mash dist genome1.txt genome2.txt` and so on:

> genome1 and genome2: I get a distance of 0.0379382
> genome3 and genome4: I get a distance of 1, whereas I should have 0.295981

[genome1.txt](https://github.com/marbl/Mash/files/3701704/genome1.txt)
[genome2.txt](https://github.com/marbl/Mash/files/3701705/genome2.txt)
[genome3.txt](https://github.com/marbl/Mash/files/3701733/genome3.txt)
[genome4.txt](https://github.com/marbl/Mash/files/3701735/genome4.txt)

**Comment by tseemann** [comment]
```
seqkit stat genome*
file         format  type  num_seqs    sum_len    min_len    avg_len    max_len
genome1.txt  FASTA   DNA          1      5,747      5,747      5,747      5,747
genome2.txt  FASTA   DNA          4     10,722          6    2,680.5      5,608
genome3.txt  FASTA   DNA          1  1,587,120  1,587,120  1,587,120  1,587,120
genome4.txt  FASTA   DNA         78  2,997,537        536     38,430    666,660
```

**Comment by aperrin** [comment]
Indeed, I forgot this information.  0.295981 is the distance I obtained before updating mash to this last version. So maybe the "real distance" is not exactly that, but it should be close.

So, when you try with mash triangle (or mash sketch for the whole matrix), you obtain the same result as me. A distance between 2 small genomes, but 1 if at least 1 of the genomes is "big". With my previous version, I had distances for all couples. Do you know why it doesn't work anymore? 

Thanks!

**Comment by tseemann** [comment]
What version did you get the result you wanted on?
Are you using 2.2 or 2.2.1 for the "bad" result?

**Comment by aperrin** [comment]
I was using version 1.1.1 to get the results with "float numbers"

I updated to version 2.2, with which I get those binary distances.

**Comment by tseemann** [comment]
FYI 2.2.2 is the latest tag, but not a 'release' per se. Nothing related to your issue has changed though. But it is faster at getting the possibly wrong answer :) https://github.com/marbl/Mash/tags .

**Comment by aperrin** [comment]
Ok. So it is not planned to correct this bug?

**Comment by tseemann** [comment]
@aperrin I am not the author of this package. I'm just a global github citizen. Ask @ondovb . I'm not even convinced its a bug. The v1 behaviour may be the bug. Your genomes are totally different in size. They should not have a good mash distance in my opinion!

**Comment by aperrin** [comment]
Ok! But I tried with 2 "big" genomes of the same size, and it does not work either. 
Anyway, thanks a lot for your help!
